src/commands/env/create/sandbox.ts

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as fs from 'fs';
+import { Flags } from '@oclif/core';
+import { SfCommand } from '@salesforce/sf-plugins-core';
+import {
+  AuthInfo,
+  Lifecycle,
+  Messages,
+  Org,
+  ResultEvent,
+  SandboxEvents,
+  SandboxProcessObject,
+  SandboxRequest,
+  SandboxUserAuthResponse,
+  SfError,
+  StatusEvent,
+} from '@salesforce/core';
+import { Duration } from '@salesforce/kit';
+import { Ux } from '@salesforce/sf-plugins-core/lib/ux';
+import { getLogSandboxProcessResult, getSandboxProgress, SandboxProgress } from '../../../shared/sandboxReporter';
+import { handleSideEffects, toKeyValuePairs } from '../../../utils';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-env', 'create.sandbox');
+const progressBarOptions = {
+  title: 'Sandbox Create',
+  format: '%s | {id} | {status} | {bar} | {percentComplete}%. {remainingWaitTimeHuman}',
+  barCompleteChar: '\u2588',
+  barIncompleteChar: '\u2591',
+  linewrap: true,
+};
+
+export default class CreateSandbox extends SfCommand<SandboxProcessObject> {
+  public static summary = messages.getMessage('summary');
+  public static description = messages.getMessage('description');
+  public static examples = messages.getMessages('examples');
+
+  public static flags = {
+    // needs to change when new flags are available
+    'definition-file': Flags.string({
+      char: 'f',
+      summary: messages.getMessage('flags.definitionFile.summary'),
+    }),
+    'set-default': Flags.boolean({
+      char: 's',
+      summary: messages.getMessage('flags.setDefault.summary'),
+    }),
+    alias: Flags.string({
+      char: 'a',
+      summary: messages.getMessage('flags.alias.summary'),
+    }),
+    wait: Flags.integer({
+      char: 'w',
+      summary: messages.getMessage('flags.wait.summary'),
+      // min: 6,
+      default: 6,
+    }),
+    'def-property': Flags.string({
+      char: 'p',
+      summary: messages.getMessage('flags.defProperty.summary'),
+      multiple: true,
+    }),
+    'target-org': Flags.string({
```

Comment:
You can use `Flags.optionalOrg` or `Flags.requiredOrg` from sf-plugins-core instead